pkg/kademlia/inspector.go

```diff
@@ -136,3 +136,29 @@ func (srv *Inspector) NodeInfo(ctx context.Context, req *pb.NodeInfoRequest) (*p
 		Version:  info.GetVersion(),
 	}, nil
 }
+
+// GetBucketList returns the list of buckets with their routing nodes and their cached nodes
+func (srv *Inspector) GetBucketList(ctx context.Context, req *pb.GetBucketListRequest) (*pb.GetBucketListResponse, error) {
+
+	bucketIds, err := srv.dht.GetBucketIds()
+	if err != nil {
+		return nil, err
+	}
+
+	buckets := make([]*pb.GetBucketListResponse_Bucket, len(bucketIds))
+
+	for i, b := range bucketIds {
+		bucketID := keyToBucketID(b)
+		routingNodes, _ := srv.dht.GetNodesWithinKBucket(bucketID)
```

Comment:
Don't ignore errors. Alternatively, when you do, add a comment why it was essential.